pages/api/account/account.js

```diff
@@ -0,0 +1,72 @@
+import logger from "@config/logger";
+import connectMongo from "@config/mongo";
+import { Account } from "@models/index";
+
+export default async function handler(req, res) {
+  const session = await getServerSession(req, res, authOptions);
+
+  if (!session) {
+    res.status(401).json({ message: "You must be logged in." });
+    return;
+  }
+
+  if (req.method !== "GET") {
+    return res.status(400).json({ error: "Invalid request: GET request required" });
+  }
+
+  res.status(200);
+}
+
+export async function getAccountByProviderAccountId(providerAccountId) {
+
+  logger.info("FUNCTION STARTED: getAccountByProviderAccountId");
+  await connectMongo();
+
+  let account = null;
+  try {
+    account = await Account.findOne({ providerAccountId });
+    if (!account) {
+      // Account not found
+      logger.info("Account not found for providerAccountId: ", providerAccountId);
+    }
+  } catch (error) {
+    logger.error("Error retrieving account:", error);
+    throw error;
+  }
+
+  logger.info("FUNCTION ENDED: getAccountByProviderAccountId");
+  return account;
+}
+
+export async function associateProfileWithAccountIfAbsentAndReturnAccount(account, newProfileId) {
+  
+  logger.info("FUNCTION STARTED: associateProfileWithAccountIfAbsentAndReturnAccount");
+
+  await connectMongo();
+ 
+  try {
+    
+    // Check if the newProfile already exists in the profiles array
+    const existingProfile = account.profiles.find((profile) => profile.equals(newProfileId));
+
+    if (existingProfile) {
+      logger.info('Profile already exists in the account. Skipping addition.' + JSON.stringify(account));
+      return account;;
+    }
+
+    // Add the newProfile to the profiles array
+    account.profiles.push(newProfileId);
+
+    // Save the updated account
+    await account.save();
```

Comment:
Can this be done in Mongo, something like...

```js
   await Account.findOneAndUpdate(
      {
        _id: providerAccountId,
        "profile._id": new ObjectId(id),
      },
      {
        $set: {
          "profiles.$": newProfile,
        },
      },
      { upsert: true, new: true }
    );
```

Reply:
I have already made a call for an account once from DB. And I think this function "findOneAndUpdate" will load the account one more time. That's the reason have passed the account from statistics.js. But I will check once.

Reply:
I have examined both `updateOne` and `findOneAndUpdate`, and they both involve making a database call to perform updates. In my code, I verify the profile mappings by checking the previously loaded account in `statistics.js`. Only if there is no match found, I make a database call to save the profile. This approach helps optimize the process by minimizing unnecessary database operations.

Reply:
No changes for this in PR  https://github.com/EddieHubCommunity/LinkFree/pull/8232 comment as I have explained in the last comment.